longReadsqc.wdl

```diff
@@ -0,0 +1,312 @@
+
+  }
+
+  call finish_rqc {
+        input: 
+          container = bbtools_container,
+          prefix = prefix,
+          filtered = bbdukReads.out_fastq,
+          pbmarkdup_stats = pbmarkdup.stats,
+          icecream_stats = icecreamfilter.stats,
+          bbdukEnds_stats = bbdukEnds.stats,
+          bbdukReads_stats = bbdukReads.stats
+  }
+
+  output {
+    File rqc_info = make_info_file.rqc_info
+    File filtered_final = finish_rqc.filtered_final
+    File filtered_stats1 = finish_rqc.filtered_stats_final
+    File filtered_stats2 = finish_rqc.filtered_stats2_final
+    File filtered_stats3 = finish_rqc.filtered_stats3_final
+    File filtered_stats4 = finish_rqc.filtered_stats4_final
+  }
+
+}
+
+task pbmarkdup{
+  input{
+    String   in_file
+    String   prefix
+    String   out_file = prefix + ".pbmarkdup.fq"
+    String?  log_level
+    Boolean? rmdup
+    Boolean? overwrite
+    String   container
+    # String   outdir
+    # String   out_file = outdir + "/pbmarkdup.fq"
+  }
+
+  command <<<
+  set -oeu pipefail
+  # mkdir -m 755 -p ~{"outdir"}
+  pbmarkdup --version 
+  pbmarkdup \
+  ~{if (defined(log_level)) then "--log-level " + log_level else  "--log-level INFO"  } \
+  ~{true="--rmdup" false="" rmdup} \
+  ~{true="--clobber" false="" overwrite} \
+  ~{in_file} \
+  ~{out_file}
+
+  gzip ~{out_file}
+
+   >>>
+
+  output{
+    File out_fastq = "~{out_file}.gz"
+    File outlog = stdout()
+    File stats = stderr()
+  }
+
+  runtime{
+     docker: container
+     continueOnReturnCode: true
+  }
+}
+
+
+task icecreamfilter{
+  input{
+    File   in_file
+    String prefix
+    String out_bad = prefix + ".icecreamfilter.out_bad.out.gz"
+    String out_good = prefix + ".icecreamfilter.out_good.out.gz"
+    String container
+    # String outdir
+    # String out_bad = outdir + "/" + prefix + ".icecreamfilter.out_bad.out.gz"
+    # String out_good = outdir + "/" + prefix + ".icecreamfilter.out_good.out.gz"
+  }
+
+  command <<<
+  set -oeu pipefail
+  icecreamfinder.sh \
+  jni=t \
+  json=t \
+  ow=t \
+  cq=f \
+  keepshortreads=f \
+  trim=f \
+  ccs=t \
+  stats=triangle.json \
+  ~{"in=" + in_file} \
+  ~{"out=" + out_good} \
+  ~{"outb=" + out_bad} 
+
+  >>>
+
+  output{
+    File output_good = "~{out_good}"
+    File output_bad = "~{out_bad}"
+    File stats = "triangle.json"
+  }
+
+  runtime{
+     docker: container
+     continueOnReturnCode: true
+  }
+}
+
+task bbdukEnds{
+  input{
+    File?  reference
+    File   in_file
+    String prefix
+    String out_file = prefix + ".bbdukEnds.out.fq.gz"
+    String container
+    # String outdir
+    # String out_file = outdir + "/" + prefix + ".bbdukEnds.out.fq.gz"
+  }
+
+  command <<<
+  set -oeu pipefail
+  # bbduk - trim out adapter from read ends
+  bbduk.sh \
+  k=20 \
+  mink=12 \
+  edist=1 \
+  mm=f \
+  ktrimtips=60 \
+  json=t \
+  ~{if (defined(reference)) then "ref=" + reference else "ref=/bbmap/resources/PacBioAdapter.fa" } \
```

Comment:
all databases should follow this convention
database="/refdata/"
within that you can specify the path further.
This is needed for JAWS compatibility. /refdata gets remapped by jaws and nmdc's automation config to /global/cfs/cdirs/m3408/refdata

Reply:
Any new database dependencies should be added to /global/cfs/cdirs/m3408/refdata/

Reply:
To my understanding, `/bbmap/resources/PacBioAdapter.fa` is part of bbtools software. The file existed in the software docker container.  Not a reference db.

Reply:
okay, lets leave then.